### ISSUE TITLE
Expose getter for ApolloClient.serverUrl

### ISSUE THREAD

**Issue by tir38** [apollographql/apollo-android#2071]
**Is your feature request related to a problem? Please describe.**
no

**Describe the solution you'd like**
I would like a simple public getter for `ApolloClient.serverUrl`. There are times where I need to rebuild the client when the serverUrl changes (our app supports dynamic URLs). But I would like to know what the "current" url is before I decide to re-build. Example

```
val initialApolloClient = ApolloClient.builder()
        .serverUrl("some URL")
        .build()

// ...some time later
val newClient = if (initialApolloClient.serverUrl == "some new URL") {
    // URLs changed, rebuild
     ApolloClient.builder()
            .serverUrl("some new URL")
            .build()
} else {
    // nothing changed, use initial client
    initialApolloClient
}
```

**Comment by tasomaniac** [comment]
Definitely a fair request. We can actually expose more. 

Would it be also useful to expose `newBuilder` function which will create a builder where you can create a new instance with `serverUrl` (or any other prop) modified. `OkHttp` has this option.

**Comment by tir38** [comment]
ohh yeah. good idea. You're thinking farther ahead than I am =)

**Comment by tasomaniac** [comment]
Merged and will be released with version 1.4.0
